src/CommunityToolkit.Maui/Converters/MathExpressionConverter/MathExpression.shared.cs

```diff
@@ -4,94 +4,144 @@
+			new ("if", 3, x => __bool(x[0]) ? x[1] : x[2]),
+
+			new ("abs", 1, x => Math.Abs(Convert.ToDouble(x[0]))),
+			new ("acos", 1, x => Math.Acos(Convert.ToDouble(x[0]))),
+			new ("asin", 1, x => Math.Asin(Convert.ToDouble(x[0]))),
+			new ("atan", 1, x => Math.Atan(Convert.ToDouble(x[0]))),
+			new ("atan2", 2, x => Math.Atan2(Convert.ToDouble(x[0]), Convert.ToDouble(x[1]))),
+			new ("ceiling", 1, x => Math.Ceiling(Convert.ToDouble(x[0]))),
+			new ("cos", 1, x => Math.Cos(Convert.ToDouble(x[0]))),
+			new ("cosh", 1, x => Math.Cosh(Convert.ToDouble(x[0]))),
+			new ("exp", 1, x => Math.Exp(Convert.ToDouble(x[0]))),
+			new ("floor", 1, x => Math.Floor(Convert.ToDouble(x[0]))),
+			new ("ieeeremainder", 2, x => Math.IEEERemainder(Convert.ToDouble(x[0]), Convert.ToDouble(x[1]))),
+			new ("log", 2, x => Math.Log(Convert.ToDouble(x[0]), Convert.ToDouble(x[1]))),
+			new ("log10", 1, x => Math.Log10(Convert.ToDouble(x[0]))),
+			new ("max", 2, x => Math.Max(Convert.ToDouble(x[0]), Convert.ToDouble(x[1]))),
+			new ("min", 2, x => Math.Min(Convert.ToDouble(x[0]), Convert.ToDouble(x[1]))),
+			new ("pow", 2, x => Math.Pow(Convert.ToDouble(x[0]), Convert.ToDouble(x[1]))),
+			new ("round", 2, x => Math.Round(Convert.ToDouble(x[0]), Convert.ToInt32(x[1]))),
+			new ("sign", 1, x => Math.Sign(Convert.ToDouble(x[0]))),
+			new ("sin", 1, x => Math.Sin(Convert.ToDouble(x[0]))),
+			new ("sinh", 1, x => Math.Sinh(Convert.ToDouble(x[0]))),
+			new ("sqrt", 1, x => Math.Sqrt(Convert.ToDouble(x[0]))),
+			new ("tan", 1, x => Math.Tan(Convert.ToDouble(x[0]))),
+			new ("tanh", 1, x => Math.Tanh(Convert.ToDouble(x[0]))),
+			new ("truncate", 1, x => Math.Truncate(Convert.ToDouble(x[0]))),
+			new ("int", 1, x => Convert.ToInt32(x[0])),
+			new ("double", 1, x => Convert.ToDouble(x[0])),
+			new ("bool", 1, x => Convert.ToBoolean(x[0])),
+			new ("str", 1, x => x[0]?.ToString()),
+			new ("len", 1, x => x[0]?.ToString()?.Length),
+			new ("^", 2, x => Math.Pow(Convert.ToDouble(x[0]), Convert.ToDouble(x[1]))),
+			new ("pi", 0, _ => Math.PI),
+			new ("e", 0, _ => Math.E),
+			new ("true", 0, _ => true),
+			new ("false", 0, _ => false),
+			new ("null", 0, _ => null),
 		};
 
 		if (argumentList.Count > 0)
 		{
-			operators.Add(new MathOperator("x", 0, MathOperatorPrecedence.Constant, _ => argumentList[0]));
+			operators.Add(new MathOperator("x", 0, _ => argumentList[0]));
 		}
 
 		for (var i = 0; i < argumentList.Count; i++)
 		{
 			var index = i;
-			operators.Add(new MathOperator($"x{i}", 0, MathOperatorPrecedence.Constant, _ => argumentList[index]));
+			operators.Add(new MathOperator($"x{i}", 0, _ => argumentList[index]));
 		}
 
 		this.operators = operators;
 		this.arguments = argumentList;
 	}
 
-	internal string Expression { get; }
+	internal string expr { get; }
+
+	internal int exprIndex { get; set; } = 0;
+
+	internal Match patternMatch { get; set; } = Match.Empty;
+
+	internal List<MathToken> rpn { get; } = new();
```

Comment:
For those can you follow the C# property name pattern?

Reply:
Hi @pictos, I would like to follow the correct C# property name pattern. I am assuming this to be necessary capitalization, e.g. `Expr`, `ExprIndex`, `PatternMatch`, `RPN`? Or did you have something else in mind?

Reply:
I would assume so. I would personally prefer full words over abbreviations e.g. ExpressionIndex over ExprIndex